addons/sale_stock/models/account_invoice.py

```diff
@@ -17,6 +18,49 @@ def _get_last_step_stock_moves(self):
             rslt += invoice.mapped('refund_invoice_id.invoice_line_ids.sale_line_ids.order_id.picking_ids.move_lines').filtered(lambda x: x.state == 'done' and x.location_id.usage == 'customer')
         return rslt
 
+    def _get_invoiced_lot_values(self):
+        """ Get and prepare data to show a table of invoiced lot on the invoice's report. """
+        self.ensure_one()
+        sale_orders = self.mapped('invoice_line_ids.sale_line_ids.order_id')
+        stock_move_lines = sale_orders.mapped('picking_ids.move_lines.move_line_ids')
+        invoice_ids = sale_orders.mapped('invoice_ids')
+        relevant_invoices = invoice_ids.filtered(lambda i: i.date_invoice < self.date_invoice)
+        last_invoice = self.env['account.invoice']
+        if relevant_invoices:
+            last_invoice = relevant_invoices.sorted(lambda i: i.date_invoice)[len(relevant_invoices) - 1]
+
+        if last_invoice:
+            incoming_sml = stock_move_lines.filtered(lambda m: m.state == 'done' and m.location_id.usage == 'customer'
+                                                          and last_invoice.date_invoice <= m.date.date() <= self.date_invoice
+                                                          and m.lot_id)
+            outgoing_sml = stock_move_lines.filtered(lambda m: m.state == 'done' and m.location_dest_id.usage == 'customer'
+                                                          and last_invoice.date_invoice <= m.date.date() <= self.date_invoice
+                                                          and m.lot_id)
+        else:
+            incoming_sml = stock_move_lines.filtered(lambda m: m.state == 'done' and m.location_id.usage == 'customer'
+                                                          and m.date.date() <= self.date_invoice
+                                                          and m.lot_id)
+            outgoing_sml = stock_move_lines.filtered(lambda m: m.state == 'done' and m.location_dest_id.usage == 'customer'
+                                                          and m.date.date() <= self.date_invoice
+                                                          and m.lot_id)
```

Comment:
we have to think of something for the customer refund, in this case it's only the quantities returned in stock that should be displayed in the table. it's probably only the qties_per_lot list to adapt
i think it's if self.type == out_refund